content/oci/oci.go

```diff
@@ -398,6 +402,49 @@ func (s *Store) writeIndexFile() error {
 	return os.WriteFile(s.indexPath, indexJSON, 0666)
 }
 
+// GC removes garbage from Store. The garbage to be cleaned are:
+//   - unreferenced (dangling) blobs in Store which have no predecessors
+//   - garbage blobs in the storage whose metadata is not stored in Store
+func (s *Store) GC(ctx context.Context) error {
+	s.sync.Lock()
+	defer s.sync.Unlock()
+
+	// clean up dangling layers in Store
+	danglings := s.graph.GetDanglingLayers()
+	for _, desc := range danglings {
+		// do not remove existing manifests in the index
+		if _, err := s.tagResolver.Resolve(ctx, string(desc.Digest)); err == errdef.ErrNotFound {
+			// remove the blob and its metadata from the Store
+			if err := s.deleteNode(ctx, desc); err != nil {
+				return err
+			}
+		}
+	}
+
+	// clean up garbage blobs in the storage
+	rootpath := filepath.Join(s.root, "blobs")
+	return filepath.Walk(rootpath, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// skip the directories
+		if !info.IsDir() {
+			alg := filepath.Base(filepath.Dir(path))
+			blobDigest, err := digest.Parse(fmt.Sprintf("%s:%s", alg, info.Name()))
+			if err != nil {
+				return err
+			}
+			if exists := s.graph.Exists(blobDigest); !exists {
+				// remove the blob from storage if it does not exist in Store
+				if err := s.storage.deleteByDigest(ctx, blobDigest); err != nil {
+					return err
+				}
```

Comment:
Since we are operating in a low level, could we call `os.Remove` directly to delete the file?

Reply:
I think it makes sense to delete a blob through the underlying storage. Maybe we can implement a `Storage.listBlobs`?

Reply:
Or maybe we can do `os.Remove` in this PR first and leave the implementation and the use of `Storage.walkBlobs` (or `ReadonlyStorage.Range`) in a separate PR? @shizhMSFT What do you think?

Reply:
SGTM

Reply:
Put `os.Remove` back.